Engine/ResourceManagement/Manager/TextureManager.cpp

```diff
@@ -15,45 +18,60 @@
 #include <IL/ilu.h>
 #include <IL/ilut.h>
 
+constexpr size_t extension_size = 3; //3 characters: DDS, TGA, JPG..
 std::shared_ptr<Texture> TextureManager::Load(uint32_t uuid, const FileData& resource_data)
 {
-	std::shared_ptr<Texture> loaded_texture;
+	std::string extension;
+	TextureOptions texture_options;
+	char* cursor = (char*)resource_data.buffer;
+	size_t bytes = sizeof(TextureOptions);
+	memcpy(&texture_options, cursor, bytes);
+
+	cursor += bytes;
+	bytes = extension_size;
+	extension.resize(extension_size);
+	memcpy( extension.data(), cursor, bytes);
+
+	size_t offset = extension_size + sizeof(TextureOptions);
 
-	if (MetafileManager::GetUUIDExportedFile(uuid).find("_normal") != std::string::npos)
```

Comment:
now the normals does not have to have "_normal" at the end?

Reply:
no, you will just need to select type normal in the importing options